packages/ui/src/components/PieChart/PieChart.tsx

```
@@ -0,0 +1,119 @@
+  total,
+  totalLabel = "Total",
+}: PieChartProps) {
+  const chartData = data.map((item, index) => ({
+    name: item.name,
+    value: item.value,
+    fill: item.color || DEFAULT_COLORS[index % DEFAULT_COLORS.length],
+  }));
+
+  const chartConfig = data.reduce(
+    (acc, item, index) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)